wgpu/src/api/device.rs

```diff
@@ -155,6 +155,21 @@ impl Device {
         ShaderModule { inner: module }
     }
 
+    /// Creates a shader module from Metal MSL shader directly.
+    ///
+    /// # Safety
+    ///
+    /// This function passes the source to the backend as-is and can potentially result in a
+    /// driver crash or bogus behaviour. No attempt is made to ensure that source code is valide.
+    #[must_use]
+    pub unsafe fn create_shader_module_msl(
+        &self,
+        desc: &ShaderModuleDescriptorMsl<'_>,
+    ) -> ShaderModule {
+        let module = unsafe { self.inner.create_shader_module_msl(desc) };
+        ShaderModule { inner: module }
+    }
```

Comment:
I think we should unify this entrypoint with `create_shader_module_spirv` and call it `create_shader_module_passthrough` which takes a `ShaderModulePassthrough` enum with each descriptor as needed.

This is cool!